lib/cartopy/tests/mpl/__init__.py

```diff
@@ -147,6 +147,8 @@ def run_figure_comparisons(self, figures, test_name):
             with filelock.FileLock(result_path + '.lock').acquire():
```

Comment:
Try instead changing this to:
```
with filelock.FileLock(result_path + '.lock'):
```
and remove the acquire. Based on this [example](https://github.com/benediktschmitt/py-filelock#examples).

Reply:
This didn't work for me. I think this is dependent on the underlying OS flock command. Apparently they will leave the file around on purpose to keep it thread safe.

I agree it would be nice to do it on exit of the context manager. Additionally, we could drop the filelock dependency and use matplotlib's own private cookbook recipe for `_path_lock`, but then we would be using a private function from them.

Reply:
We could try `flufl.lock`; it doesn't seem to leave the lock file around.

Reply:
Just tried it out locally and seems to be fine there. Lets see what CI thinks.